vowpalwabbit/core/include/vw/core/learner.h

```diff
@@ -388,10 +388,12 @@ class learner
   }
 
   // called when metrics is enabled.  Autorecursive.
-  void NO_SANITIZE_UNDEFINED persist_metrics(metric_sink& metrics)
+  void NO_SANITIZE_UNDEFINED persist_metrics(VW::metrics_manager& manager)
   {
-    _persist_metrics_fd.save_metric_f(_persist_metrics_fd.data, metrics);
-    if (_persist_metrics_fd.base) { _persist_metrics_fd.base->persist_metrics(metrics); }
+    auto metric_f = [this](metric_sink& sink) -> void
+    { _persist_metrics_fd.save_metric_f(_persist_metrics_fd.data, sink); };
+    manager.register_metrics_callback(metric_f);
+    if (_persist_metrics_fd.base) { _persist_metrics_fd.base->persist_metrics(manager); }
```

Comment:
Why change this function? Based on the current design it seems like the metric manager shouldn't be coupled with calling persist_metrics but just consume the output of it.

Reply:
reverted